store/sqlstore/thread_store.go

```diff
@@ -624,13 +599,21 @@ func (s *SqlThreadStore) MarkAllAsReadByChannels(userID string, channelIDs []str
 func (s *SqlThreadStore) MarkAllAsRead(userId string, threadIds []string) error {
 	timestamp := model.GetMillis()
 
-	query, args, err := s.getQueryBuilder().
-		Update("ThreadMemberships").
-		Where(sq.Eq{"UserId": userId}).
-		Where(sq.Eq{"PostId": threadIds}).
+	var querySQL sq.UpdateBuilder
+	if s.DriverName() == model.DatabaseDriverPostgres {
+		querySQL = s.getQueryBuilder().Update("ThreadMemberships").From("Threads")
+
+	} else {
+		querySQL = s.getQueryBuilder().Update("ThreadMemberships", "Threads")
+	}
+
+	query, args, err := querySQL.
 		Set("LastViewed", timestamp).
 		Set("UnreadMentions", 0).
 		Set("LastUpdated", model.GetMillis()).
+		Set("SeenReplyCount", "Threads.ReplyCount").
```

Comment:
Are we missing something to effectively JOIN `ThreadMemberships` and `Threads` together vs. creating a cross join?